prow/config/config.go

```diff
@@ -757,9 +757,17 @@ type DefaultDecorationConfigEntry struct {
 // TODO(mpherman): Make a Matcher struct embedded in both ProwJobDefaultEntry and
 // DefaultDecorationConfigEntry and DefaultRerunAuthConfigEntry.
 func matches(givenOrgRepo, givenCluster, orgRepo, cluster string) bool {
-	orgRepoMatch := givenOrgRepo == "" || givenOrgRepo == "*" || givenOrgRepo == strings.Split(orgRepo, "/")[0] || givenOrgRepo == orgRepo
-	clusterMatch := givenCluster == "" || givenCluster == "*" || givenCluster == cluster
-	return orgRepoMatch && clusterMatch
+	if givenCluster != "" && givenCluster != "*" && givenCluster != cluster {
```

Comment:
I can't claim to understand the previous boolean algebra enough to understand this one. But I think the new style here is far more readable.